ext/dcos-installer/dcos_installer/action_lib/__init__.py

```diff
@@ -65,7 +78,7 @@ def run_preflight(config, pf_script_path='/genconf/serve/dcos_install.sh', block
     chains.append(cleanup_chain)
     master_agent_count = {
         'total_masters': len(config['master_list']),
-        'total_agents': len(config['agent_list'])
+        'total_agents': len(config['agent_list']) + len(config['public_agent_list'])
```

Comment:
This code comes up twice. Why not have a single function which you call to go from config -> nodes_count_by_type?

This could also potentially be more cleanly written using a python groupby (https://docs.python.org/3/library/itertools.html#itertools.groupby). But that is probably out of scope for this PR.